tools/donate_cpu_lib.py

```diff
@@ -42,12 +42,19 @@ def get_cppcheck(cppcheck_path, work_path):
             try:
                 os.chdir(cppcheck_path)
                 try:
+                    subprocess.check_call(['git', 'fetch'])
```

Comment:
Would it make sense to limit this to just the branch in question?

Is there a way to tell if `git fetch` actually did something and thus completely eliminating the `git rev-parse` calls?

Reply:
I don't understand what you mean to limiting git fetch to just the branch. Could you elaborate which code branch ? 

Ideally we'd want "git fetch" to tell us somehow whether there are new commits. I looked there first, but searching docs and googling turned up nothing. I also looked at the exit code, it is the same regardless of whether there's new commit or not. 
One alternative could be to parse "git fetch" stdout, but string parsing would make this fragile and may break depending of git version and system locale. 
The best alternative I could find is rev-parse to get hashes. This adds a couple git invocation, but it's well worth it given we get a nice efficiency gain overall.

Reply:
Oh, you meant git branch, not code branch! 
I will test "git fetch main" and update the merge request based on the result.